packages/@react-stately/data/src/useListData.ts

```diff
@@ -152,9 +152,28 @@ export function useListData<T>(options: ListOptions<T>): ListData<T> {
     filterText: initialFilterText
   });
 
+  const filterItems = (items: T[], filterText: string): T[] => {
+    return items.reduce((acc: T[], item: any) => {
+      if (item.children) {
+        if (filter(item, filterText)) {
+          acc.push(item);
```

Comment:
Question to team, should we do section name matching?

I'm of the opinion that only items themselves should be matched, as you cannot select a section and this is meant to be a way to narrow down the options.

Reply:
I think there is an argument for section matching in that some designs may rely on sections as the important/distinct entities. For instance, a collection with many unique sections, and each section having a small number of items that are the same in each section.

I think this would warrant an additional argument to customize this behavior, and we could decide what makes the most sense for the default behavior.

Reply:
Hmm I think section name matching should be possible since people might want to just filter by sections and then select an item within that particular section.